<?xml version="1.0" encoding="utf-8"?>

<launch>
  
  <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
  
  <!-- Bag file path -->  
  <arg name="bag_file" value="$(env HOME)/.segmap/kitti/2011_10_03_drive_27.bag"/>
	
  <!-- Rviz visualization config file path -->
  <arg name="rviz_config_file" value="$(env HOME)/.segmap/kitti/kitti_localization.rviz"/>

  <!-- ROS parameters -->
  <rosparam>use_sim_time: true</rosparam>

  <rosparam command="load" file="$(find segmapper)/launch/kitti/kitti_localization.yaml"/>
  
  <!-- Target cloud -->
  <arg name="target_cloud" value="$(env HOME)/.segmap/kitti/drive27_target_map.pcd"/>
  <rosparam param="/SegMapper/SegMatchWorker/target_cloud_filename" subst_value="True">$(arg target_cloud)</rosparam>
  
  <!-- Callgrind profiling option -->
  <arg name="enable_callgrind" default="false" />
  <arg name="play_rate" value="0.005" if="$(arg enable_callgrind)"/>
  <arg name="play_rate" value="1" unless="$(arg enable_callgrind)"/>

  <!-- rviz node -->
  <node name="visualizer" pkg="rviz" type="rviz" output="screen" respawn="$(arg respawn_value)" args="-d $(arg rviz_config_file)" />
  
  <!-- rosbag player node -->
  <node name="player" pkg="rosbag" type="play" output="screen" args="-r $(arg play_rate) -s 333 --clock --pause $(arg bag_file)"/>

  <!-- segmapper node -->
  <include file="$(find segmapper)/launch/segmapper.launch">
    <arg name="icp_configuration_file" value="$(find segmapper)/launch/kitti/icp_dynamic_outdoor.yaml"/>
    <arg name="icp_input_filters_file" value="$(find segmapper)/launch/kitti/input_filters_outdoor.yaml"/>
    <arg name="enable_callgrind" value="$(arg enable_callgrind)"/>
  </include>
	
</launch>
